quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/configuration/mappers/WildcardPropertyMapper.java

```diff
@@ -23,9 +26,10 @@ public class WildcardPropertyMapper<T> extends PropertyMapper<T> {
 
     private static final Pattern valueValidator = Pattern.compile("[\\[\\]\\$\\-._a-zA-Z0-9]+");
 
+    private final Supplier<List<WildcardPropertyMapper<?>>> connectedMappers;
```

Comment:
We no longer need the connectedMappers concept correct? We just need to ensure a single runtime option is present.

Reply:
Not exactly like that. It is true that we'd need to ensure a single runtime option is present, but it's not the end. For additional datasource, we need to also advertise transaction settings, otherwise the datasource will not work as expected. 

I find the connectedMappers functionality suitable for these cases as we advertise the connected options (with the same wildcard key) that have default values + the required JDBC URL prop. It prevents to face some issues in the future that we might forget to explicitly advertise something related.

I think this functionality might be useful even for the Hibernate ORM wildcard properties.

@shawkins Does it work for you, or do you suggest a different approach?

Reply:
> Could we just explicitly list connected options (URL + txn) as part of the option builder for DB kind? Then we shouldn't need a specific check for db kind in the PropertyMappingInterceptor - just whether the wildcard has connected options.

+1